### PR TITLE
feat: Convert singlesig addr to multisig-1of1 addr

### PR DESCRIPTION
**Pull request by keroro520**
```
ckb-cli util single-to-multisig-addr --address ckb1qyq8nry7w47zqmzqc8r7shk2fut0pz8w072qht72mw --locktime 2019-11-22
```

```
"ckb1qyq8nry7w47zqmzqc8r7shk2fut0pz8w072qht72mw,2019-11-22,ckb1qyqszqgpq9uce8n40ssxcsxpcl59ajj0zmcg3mnljsygchrd,2307822145243185186"
```

### CODE REVIEW

**Comment by keroro520** [comment]
```
$ cat 1.txt

ckb1qyq8nry7w47zqmzqc8r7shk2fut0pz8w072qht72mw,2019-11-02
ckb1qyq8nry7w47zqmzqc8r7shk2fut0pz8w072qht72mw,2019-11-01
ckb1qyqtd3u2pm0umlequeqcydwl7czgm8en6swsejam8z,2020-11-21

$ cat 1.txt | awk -F ',' '{ system("./target/debug/ckb-cli util single-to-multisig-addr --address " $1 " --locktime " $2) }'

"ckb1qyq8nry7w47zqmzqc8r7shk2fut0pz8w072qht72mw,2019-11-02,ckb1qyqszqgpq9uce8n40ssxcsxpcl59ajj0zmcg3mnljsygchrd,2305844108725321728"
"ckb1qyq8nry7w47zqmzqc8r7shk2fut0pz8w072qht72mw,2019-11-01,ckb1qyqszqgpq9uce8n40ssxcsxpcl59ajj0zmcg3mnljsygchrd,2305844108725321728"
"ckb1qyqtd3u2pm0umlequeqcydwl7czgm8en6swsejam8z,2020-11-21,ckb1qyqszqgpqxmv0zswmlxl7g8xgxprthlkqjxe7v75r527tmpv,2307822130143692978"
```

File: ckb-sdk/src/basic.rs [hunk 1]
```diff
@@ -153,6 +154,52 @@ impl Address {
...
+        let network = NetworkType::MainNet;
+        let hrp = network.to_prefix();
+        let mut data = [0u8; 26];
+        data[0] = self.ty as u8;
+        data[1] = CodeHashIndex::Multisig as u8;
+        data[2] = 1; // S
+        data[3] = 1; // R
+        data[4] = 1; // M
+        data[5] = 1; // N
+        data[6..26].copy_from_slice(self.hash.as_bytes());
```
Comment (TheWaWaR) [inline ckb-sdk/src/basic.rs]: The multisig script hash is the 20 bytes blake160 **hash** of multisig script.
https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0021-ckb-address-format/0021-ckb-address-format.md#short-payload-format

`multisig` argument is not included in address.

File: ckb-sdk/src/basic.rs [hunk 2]
```diff
@@ -153,6 +154,52 @@ impl Address {
...
         format!("{}", value)
     }
+
+    pub fn to_multi_1of1_string(&self) -> String {
+        let network = NetworkType::MainNet;
+        let hrp = network.to_prefix();
+        let mut data = [0u8; 26];
+        data[0] = self.ty as u8;
+        data[1] = CodeHashIndex::Multisig as u8;
+        data[2] = 1; // S
```
Comment (doitian) [inline ckb-sdk/src/basic.rs]:
```suggestion
        data[2] = 0; // S
```

File: ckb-sdk/src/basic.rs [hunk 3]
```diff
@@ -153,6 +154,52 @@ impl Address {
...
+        data[3] = 1; // R
```
Comment (doitian) [inline ckb-sdk/src/basic.rs]:
```suggestion
        data[3] = 0; // R
```

File: ckb-sdk/src/basic.rs [hunk 4]
```diff
@@ -153,6 +154,52 @@ impl Address {
...
+        data[4] = 1; // M
+        data[5] = 1; // N
+        data[6..26].copy_from_slice(self.hash.as_bytes());
+        let value = Bech32::new(hrp.to_string(), data.to_base32())
+            .unwrap_or_else(|_| panic!("Encode address failed: hash={:?}", self.hash));
+        format!("{}", value)
+    }
+
+    pub fn from_multi_1of1_string(input: &str) -> Result<Self, String> {
```
Comment (doitian) [inline ckb-sdk/src/basic.rs]: From the string, we cannot restore public key hash.

**Pull request closed without merging**